src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/Initializers/WebJobsTelemetryInitializer.cs

```diff
@@ -122,5 +129,52 @@ private static string GetRoleInstanceName()
 
             return instanceName;
         }
+
+        /// <summary>
+        /// Changes properties of the RequestTelemetry to match what Functions expects.
+        /// </summary>
+        /// <param name="request">The RequestTelemetry to update.</param>
+        private void UpdateRequestProperties(RequestTelemetry request)
+        {
+            request.Context.GetInternalContext().SdkVersion = _sdkVersion;
+
+            // If the code hasn't been set, it's not an HttpRequest (could be auto-tracked SB, etc).
+            // So we'll initialize it to 0
+            if (string.IsNullOrEmpty(request.ResponseCode))
+            {
+                request.ResponseCode = "0";
+            }
+        }
+
+        /// <summary>
+        /// Tries to apply well-known properties from a KeyValuePair onto the RequestTelemetry.
+        /// </summary>
+        /// <param name="request">The request.</param>
+        /// <param name="property">The property.</param>
+        /// <returns>True if the property was applied. Otherwise, false.</returns>
+        private bool TryApplyProperty(RequestTelemetry request, KeyValuePair<string, string> property)
+        {
+            bool wasPropertySet = false;
+
+            if (property.Key == LogConstants.NameKey)
+            {
+                request.Context.Operation.Name = property.Value;
+                request.Name = property.Value;
+                wasPropertySet = true;
+            }
+            else if (property.Key == LogConstants.SucceededKey &&
+                bool.TryParse(property.Value, out bool success))
+            {
+                // no matter what App Insights says about the response, we always
+                // want to use the function's result for Succeeeded
+                request.Success = success;
+                wasPropertySet = true;
+
+                // Remove the Succeeded property as it's duplicated
+                request.Properties.Remove(LogConstants.SucceededKey);
+            }
```

Comment:
FYI -- this part where we check the SucceededKey is the major change. It will be needed for Http as well.

Reply:
I wonder if we can prevent it from being set on properties at the first place. It's set in the AppInsightsLogger. but now it's needed because of https://github.com/Microsoft/ApplicationInsights-dotnet-server/issues/1038 anyway